src/types.ts

```diff
@@ -1,8 +1,16 @@
-import { PieceType, Square } from '../cjsmin/src/chess';
+import { Square, UASymbol } from '../cjsmin/src/chess';
 
 export interface FileReaderGame {
   moves: string;
   metadata: string[];
 }
 
+export type UAPMap<T> = {
+  [key in UASymbol]: T;
```

Comment:
T as the key? to set it to something that wouldn't be used as starting value?

Reply:
look up generic types